packages/server/lib/cloud/api/get_app_studio.ts

```diff
@@ -0,0 +1,123 @@
+import path from 'path'
+import os from 'os'
+import { ensureDir, copy, readFile } from 'fs-extra'
+import cloudApi from '.'
+import { StudioManager } from '../studio'
+import tar from 'tar'
+import { verifySignatureFromFile } from '../encryption'
+import crypto from 'crypto'
+import fs from 'fs'
+import fetch from 'cross-fetch'
+import { agent } from '@packages/network'
+import { asyncRetry, linearDelay } from '../../util/async_retry'
+import { isRetryableError } from '../network/is_retryable_error'
+
+const routes = require('../routes')
+
+const _delay = linearDelay(500)
+
+const studioPath = path.join(os.tmpdir(), 'cypress', 'studio')
+const bundlePath = path.join(studioPath, 'bundle.tar')
+const serverFilePath = path.join(studioPath, 'server', 'index.js')
+
+const downloadAppStudioBundleToTempDirectory = async (projectId?: string): Promise<void> => {
+  let responseSignature: string | null = null
+
+  await (asyncRetry(async () => {
+    const response = await fetch(routes.apiRoutes.studio() as string, {
+      // @ts-expect-error - this is supported
+      agent,
+      method: 'GET',
+      headers: {
+        'x-route-version': '1',
+        'x-cypress-signature': cloudApi.publicKeyVersion,
+        ...(projectId ? { 'x-cypress-project-slug': projectId } : {}),
+        'x-cypress-studio-mount-version': '1',
+      },
+      encrypt: 'signed',
+    })
+
+    responseSignature = response.headers.get('x-cypress-signature')
+
+    await new Promise<void>((resolve, reject) => {
+      const writeStream = fs.createWriteStream(bundlePath)
+
+      writeStream.on('error', reject)
+      writeStream.on('finish', () => {
+        resolve()
+      })
+
+      // @ts-expect-error - this is supported
+      response.body?.pipe(writeStream)
+    })
+  }, {
+    maxAttempts: 3,
+    retryDelay: _delay,
+    shouldRetry: isRetryableError,
+  }))()
+
+  if (!responseSignature) {
+    throw new Error('Unable to get studio signature')
+  }
+
+  const verified = await verifySignatureFromFile(bundlePath, responseSignature)
+
+  if (!verified) {
+    throw new Error('Unable to verify studio signature')
+  }
+}
+
+const getTarHash = async (): Promise<string> => {
+  let hash = ''
+
+  return new Promise<string>((resolve, reject) => {
+    fs.createReadStream(bundlePath)
+    .pipe(crypto.createHash('sha256'))
+    .setEncoding('base64url')
+    .on('data', (data) => {
+      hash += String(data)
+    })
+    .on('error', reject)
+    .on('close', () => {
+      resolve(hash)
+    })
+  })
+}
+
+export const getAppStudio = async (projectId?: string): Promise<StudioManager> => {
+  try {
+    let script: string
+    let studioHash: string | undefined
+
+    // First remove studioPath to ensure we have a clean slate
+    await fs.promises.rm(studioPath, { recursive: true, force: true })
+    await ensureDir(studioPath)
+
+    if (process.env.CYPRESS_LOCAL_STUDIO_PATH) {
```

Comment:
Add a comment that this is removed in the binary.

Reply:
Added: [`634109a` (#31033)](https://github.com/cypress-io/cypress/pull/31033/commits/634109a70223aa69af77b1fa5415ce3891556128)